src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/base/CheckboxesContainerComponent.tsx

```diff
@@ -117,11 +117,12 @@ export const CheckboxContainerComponent = (props: ICheckboxContainerProps) => {
 
   const onDataChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
     const newSelected: any = selected.slice();
+    const index = newSelected.indexOf(event.target.name);
 
-    if (newSelected[event.target.value] === event.target.name) {
-      newSelected[event.target.value] = '';
+    if (index >= 0) {
+      newSelected[index] = '';
     } else {
-      newSelected[event.target.value] = event.target.name;
+      newSelected[newSelected.length] = event.target.name;
```

Comment:
This change means that the order of the options in the form data is not consistent, it will depend on the order the end user selects each item, instead of the order they are listed in. Not sure this is a problem at all though. 
Since we are just adding the newly selected value to the end of the array here, we could just as well use `push`:
```suggestion
      newSelected.push(event.target.name);
```

Reply:
I guess that is what the previous code was attempting to do, keep the order the same every time, 
but `newSelected[event.target.value]` didn't make sense as new selected might only have one element and event.target.value would be out of range, hence a match was never found to remove the element from the list. 

I guess we could sort the list based on their idexes somehow, but that seems like a lot of work. Maybe we can check with the app owner's if the order is essential or not?

Reply:
The more I think about it, the more unnecessary it seems to keep this order consistent. I think we should leave it as you implemented it, and revisit if someone actually can present a case for order being important.